knowledge_graph/routes.py

```diff
@@ -119,45 +119,59 @@ def receive_user_scores() -> Tuple[Response, int]:
     POSITIVITY_CONSTANT = 3.5
     RESPONSES_TO_ADD = 10
 
-    # todo: I have a horrible feeling this is wrong
-    # and will collide. fix soon.
     session_id = str(uuid.uuid4())
 
-    for value in parameter["SetOne"]:
-        questionID = value["id"]
-        score = value["score"]
+    questions = parameter["questionResponses"]
+
+    if len(questions["SetOne"]) < RESPONSES_TO_ADD:
+        return make_response("not enough set one scores", 400)
+    elif len(questions["SetOne"]) > RESPONSES_TO_ADD:
+        return make_response("too many set one scores", 400)
+
+    for value in questions["SetOne"]:
+        questionID = value["questionId"]
+        score = value["answerId"]
         overall_sum += score
+
+        if value_id_map[questionID] in value_scores:
+            return make_response("duplicate question ID", 400)
+
         value_scores[value_id_map[questionID]] = score
 
-    if parameter["SetTwo"]:
+    if "SetTwo" in questions:
         num_of_responses += RESPONSES_TO_ADD
-        for value in parameter["SetTwo"]:
-            questionID = value["id"]
-            score = value["score"]
+        for value in questions["SetTwo"]:
+            questionID = value["questionId"]
+            score = value["answerId"]
             name = value_id_map[questionID]
             avg_score = (value_scores[name] + score) / NUMBER_OF_SETS
             overall_sum += score
+
+            if questionID in value_scores.keys:
+                return make_response("duplicate question ID", 400)
+
             value_scores[name] = avg_score
 
     overall_avg = overall_sum / num_of_responses
 
     for value, score in value_scores.items():
-
         centered_score = (
             score - overall_avg + POSITIVITY_CONSTANT
         )  # To make non-negative
 
         value_scores[value] = centered_score
 
-    value_scores["session-id"] = str(session_id)
+    value_scores["session-id"] = session_id
 
     try:
         persist_scores(value_scores)
     except KeyError:
         return make_response("invalid key"), 400
 
-    response = Response(dumps(value_scores))
-    return response, 200
+    response = {"session-id": session_id}
+
+    response = Response(dumps(response))
+    return response, 201
 
 
 @app.route("/personal_values", methods=["GET"])
```

Comment:
Probably need to change the session-id for the "/personal_values" endpoint to not be int() but str()